### PR TITLE
MAINT: Remove numpy/npy_3kcompat.h from _superluobject.c

### PR DESCRIPTION
**Pull request by sethtroisi**
#### Reference issue

Continued work to remove all Python2 Code #11584

#### What does this implement/fix?

Removes use of [`numpy/npy_3kcompat.h`](
https://github.com/numpy/numpy/blob/master/numpy/core/include/numpy/npy_3kcompat.h) in SuperLU module init.

The non-whitespace change is `s/PyInt_Check/PyLong_Check/` and `s/PyInt_AsLong/PyLong_AsLong`, I'm happy to break out the retab into a seperate PR if that would aid clarity.

PyInt_Check also does overflow checking which PyLong_Check doesn't but in this case (were the values are options/enums this is probably fine.

### CODE REVIEW

File: scipy/sparse/linalg/dsolve/_superluobject.c [hunk 1]
```diff
@@ -819,8 +818,8 @@ PyObject *newSuperLUObject(SuperMatrix * A, PyObject * option_dict,
         if (tmpobj == NULL) return 0;           \
         s = PyBytes_AS_STRING(tmpobj);         \
     }                                           \
-    else if (PyInt_Check(input)) {              \
-        i = PyInt_AsLong(input);                \
+    else if (PyLong_Check(input)) {              \
+        i = PyLong_AsLong(input);                \
```
Comment (sethtroisi) [inline scipy/sparse/linalg/dsolve/_superluobject.c:822]: 1/3 of the non-whitespace change

File: scipy/sparse/linalg/dsolve/_superluobject.c [hunk 2]
```diff
@@ -1001,48 +1000,48 @@ static int droprule_cvt(PyObject * input, int *value)
...
-	return 1;
+        /* Leave as default */
+        return 1;
     }
-    else if (PyInt_Check(input)) {
-	*value = PyInt_AsLong(input);
-	return 1;
+    else if (PyLong_Check(input)) {
+        *value = PyLong_AsLong(input);
+        return 1;
```
Comment (sethtroisi) [inline scipy/sparse/linalg/dsolve/_superluobject.c:1008]: 1/3 of the non-whitespace change

File: scipy/sparse/linalg/dsolve/_superluobject.c [hunk 3]
```diff
@@ -1057,45 +1056,45 @@ static int droprule_cvt(PyObject * input, int *value)
...
     return 1;
 }
 
 static int int_cvt(PyObject * input, int *value)
 {
     if (input == Py_None)
-	return 1;
-    *value = PyInt_AsLong(input);
+        return 1;
+    *value = PyLong_AsLong(input);
```
Comment (sethtroisi) [inline scipy/sparse/linalg/dsolve/_superluobject.c:1070]: Last non whitespace change

**Comment by tylerjereddy** [comment]
>  I'm happy to break out the retab into a seperate PR if that would aid clarity.

The diff goes from ~5 lines to ~200 lines changed with the whitespace included. Why do you think changing the whitespace is helpful at all? C isn't sensitive to whitespace, right? And there's only 1 C file modified here.

**Comment by sethtroisi** [comment]
> > I'm happy to break out the retab into a seperate PR if that would aid clarity.
> 
> The diff goes from ~5 lines to ~200 lines changed with the whitespace included. Why do you think changing the whitespace is helpful at all? C isn't sensitive to whitespace, right? And there's only 1 C file modified here.

The tabs are used inconsistently so that the file is hard to read with both ts=4 and ts=8.
which lead to this in several places
```
    if (test)
    return val1
    else {
        val2 = val1 + 1;
        return val2;
    }    
```

As I said I'm happy to break that out as a first change, not sure how maintainers feel about whitespace only changes.

**Comment by tylerjereddy** [comment]
I think I'm -0.5 on the whitespace changes either here or separately.

**Comment by rgommers** [comment]
> As I said I'm happy to break that out as a first change, not sure how maintainers feel about whitespace only changes.

Our rule of thumb has been that if you're working on some piece of code, it's fine to include cleanups to that piece of code. And we prefer no pure style cleanup PRs. In this case it's kind of borderline; including with this PR was definitely better than separate.

I agree that the indentation is weird here. The downside is that it makes git blame harder to use. I'd lean towards merging as is, but not do these kinds of large-scale re-indentings for future PRs.

I'll leave it open for a day or two in case there's objections.

**Comment by rgommers** [comment]
Okay no further comments, so merged. Thanks @sethtroisi

**Pull request merged**